src/test/java/com/shapesecurity/shift/es2016/test262/Test262.java

```diff
@@ -0,0 +1,196 @@
+		boolean onlyStrict = false;
+		boolean async = false;
+		boolean module = false;
+		if (rawFlags != null) {
+			ArrayList<String> flags = (ArrayList<String>) rawFlags;
+			for (String flag : flags) {
+				switch (flag) {
+					case "noStrict":
+						noStrict = true;
+						break;
+					case "onlyStrict":
+						onlyStrict = true;
+						break;
+					case "async":
+						async = true;
+						break;
+					case "module":
+						module = true;
+						break;
+				}
+			}
+		}
+		ImmutableSet<String> featureSet = ImmutableSet.emptyUsingEquality();
+		Object rawFeatures = parsedYaml.get("features");
+		if (rawFeatures != null) {
+			ArrayList<String> features = (ArrayList<String>) rawFeatures;
+			for (String feature : features) {
+				featureSet = featureSet.put(feature);
+			}
+		}
+		return new Test262Info(path, negativeEnum, noStrict, onlyStrict, async, module, featureSet);
+	}
+
+	private static final class Test262Exception extends RuntimeException {
+
+		@Nonnull
+		public final String name;
+
+		public Test262Exception(@Nonnull String name, @Nonnull String message) {
+			super(message);
+			this.name = name;
+		}
+
+		public Test262Exception(@Nonnull String name, @Nonnull String message, @Nonnull Throwable caused) {
+			super(message, caused);
+			this.name = name;
+		}
+	}
+
+	private void runTest262Test(@Nonnull String source, @Nonnull Path path, @Nonnull Test262Info info, boolean strict) {
+		boolean shouldFail = info.negative == Test262Info.Test262Negative.PARSE || info.negative == Test262Info.Test262Negative.EARLY;
+		boolean xfailed = XFailHelper.isXFailed(info, shouldFail);
+		boolean failed;
+		Throwable failureReason;
+		try {
+			Program program;
+			if (info.module) {
+				program = Parser.parseModule(source);
+			} else if (strict) {
+				program = Parser.parseScript("\"use strict\";\n" + source);
+			} else {
+				program = Parser.parseScript(source);
+			}
+			ImmutableList<EarlyError> earlyErrors = EarlyErrorChecker.validate(program);
+			failed = earlyErrors.length > 0;
+			failureReason = new RuntimeException("Early Errors: \n" + earlyErrors.foldLeft((str, error) -> error.message + ", " + str, ""));
+		} catch (JsError e) {
+			failed = true;
+			failureReason = e;
+		}
+		if (xfailed && failed == shouldFail || !xfailed && failed != shouldFail) {
+			if (shouldFail != xfailed) {
+				throw new Test262Exception(info.name, "Parsed and should not have: " + path.toString());
+			} else {
+				throw new Test262Exception(info.name, "Did not parse and should have: " + path.toString(), failureReason);
+			}
+		}
+	}
+
+	private void runTest(@Nonnull Path root, @Nonnull Path path) throws IOException {
+		if (Files.isDirectory(path) || !path.toString().endsWith(".js")) {
+			return;
+		}
+		String source = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);
+		Test262Info info = extractTest262Info(root.relativize(path).toString(), source);
+		if (info == null) { // parse failure, probably a fixture
```

Comment:
I would prefer we explicitly skip fixtures, and then require the frontmatter to parse. That way we don't accidentally pass a test just because its frontmatter is in a format we don't recognize.

Per [INTERPRETING.md](https://github.com/tc39/test262/blob/dec4e5df7481daf086036a152abb30492dec398a/INTERPRETING.md#modules), fixtures are those which end with `_FIXTURE.js`. We can just ignore those files.